client/components/data/query-site-settings/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingSiteSettings } from 'state/site-settings/selectors';
+import { requestSiteSettings } from 'state/site-settings/actions';
+
+class QuerySiteSettings extends Component {
+	componentWillMount() {
+		const { requestingSiteSettings, siteId } = this.props;
+		if ( ! requestingSiteSettings && siteId ) {
+			this.props.requestSiteSettings( siteId );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { siteId } = this.props;
+		if ( ! nextProps.siteId || siteId === nextProps.siteId ) {
+			return;
+		}
+
+		nextProps.requestSiteSettings( nextProps.siteId );
```

Comment:
What if a request is already in progress? Missing a similar check to that in `componentWillMount`.